api/apiclient_test.go

```diff
@@ -153,14 +154,91 @@ func (s *apiclientSuite) TestServerRoot(c *gc.C) {
 }
 
 func (s *apiclientSuite) TestDialWebsocketStopped(c *gc.C) {
+	f := (*api.NewWebsocketDialer)(nil, api.DialOpts{})
 	stopped := make(chan struct{})
-	f := api.NewWebsocketDialer(nil, api.DialOpts{})
 	close(stopped)
 	result, err := f(stopped)
 	c.Assert(err, gc.Equals, parallel.ErrStopped)
 	c.Assert(result, gc.IsNil)
 }
 
+func (s *apiclientSuite) TestOpenWithSNIHostNameEmptyCert(c *gc.C) {
```

Comment:
Sorry, @rogpeppe. It's probably because I don't know how to use GitHub reviews yet, but did this name change? The name I see is `TestOpenWithSNIHostNameEmptyCert` which clearly enumerates what's being tested, but gives me no clue what behavior I should expect.

Same comment for the other tests.